webapp/tests/karma/ts/services/create-user-for-contacts.service.spec.ts

```diff
@@ -0,0 +1,373 @@
+
+      expect(userContact).to.equal(ORIGINAL_CONTACT);
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(1);
+      expect(medicDb.get.callCount).to.equal(1);
+      expect(medicDb.get.args[0]).to.deep.equal([ORIGINAL_CONTACT._id]);
+    });
+
+    it('returns undefined if the user has no contact id', async () => {
+      userSettingsService.get.resolves({ });
+
+      const userContact = await service.getUserContact();
+
+      expect(userContact).to.be.undefined;
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(0);
+      expect(medicDb.get.callCount).to.equal(0);
+    });
+
+    it(`returns undefined if the user's contact id is not associated with any document`, async () => {
+      userSettingsService.get.resolves({ contact_id: ORIGINAL_CONTACT._id });
+      medicDb.get.withArgs(ORIGINAL_CONTACT._id).rejects({ status: 404 });
+
+      const userContact = await service.getUserContact();
+
+      expect(userContact).to.be.undefined;
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(1);
+      expect(medicDb.get.callCount).to.equal(1);
+      expect(medicDb.get.args[0]).to.deep.equal([ORIGINAL_CONTACT._id]);
+    });
+
+    it(`throws an error is an error is encountered getting the user's contact`, async () => {
+      userSettingsService.get.resolves({ contact_id: ORIGINAL_CONTACT._id });
+      medicDb.get.withArgs(ORIGINAL_CONTACT._id).rejects({ message: 'Server Error' });
+
+      try {
+        await service.getUserContact();
+        expect.fail('should have thrown an error');
+      } catch (err) {
+        expect(err.message).to.equal('Server Error');
+      }
+
+      expect(userSettingsService.get.callCount).to.equal(1);
+      expect(dbService.get.callCount).to.equal(1);
+      expect(medicDb.get.callCount).to.equal(1);
+      expect(medicDb.get.args[0]).to.deep.equal([ORIGINAL_CONTACT._id]);
+    });
+  });
+
+  describe('setReplaced', () => {
+    it('sets the given new contact and a status of PENDING when the user is offline', () => {
+      const originalContact = Object.assign({}, ORIGINAL_CONTACT);
+
+      service.setReplaced(originalContact, NEW_CONTACT);
+
+      expect(originalContact.user_for_contact.replaced).to.deep.equal({
+        status: 'PENDING',
+        by: NEW_CONTACT._id
+      });
+    });
+
+    it('sets the given new contact and a status of READY when the user is online', () => {
+      const originalContact = Object.assign({}, ORIGINAL_CONTACT);
+      sessionService.isOnlineOnly.returns(true);
+
+      service.setReplaced(originalContact, NEW_CONTACT);
+
+      expect(originalContact.user_for_contact.replaced).to.deep.equal({
+        status: 'READY',
+        by: NEW_CONTACT._id
+      });
+    });
+
+    it('sets the given new contact and status when the contact already has user_for_contact data', () => {
```

Comment:
Please add a test when a user already has a `user_for_contact.replaced` property.

Reply:
I believe that case is covered in [this existing test](https://github.com/medic/cht-core/blob/b43d13f92c34707ef92d6dacaab936e35c71e9e8/webapp/tests/karma/ts/services/create-user-for-contacts.service.spec.ts#L325).